prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -70,6 +75,39 @@ class ReportFunction {
         ) request: HttpRequestMessage<String?>,
         context: ExecutionContext,
     ): HttpResponseMessage {
+        return ingestReport(request, context)
+    }
+
+    /**
+     * POST a report to the router, using FHIR auth security
+     * This one is "/api/report".  The other one is "/api/reports"
+     */
+    @FunctionName("report")
+    @StorageAccount("AzureWebJobsStorage")
+    fun report(
+        @HttpTrigger(
+            name = "reqWithFHIRAuth",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        val workflowEngine = WorkflowEngine()
+        val tokenAuthentication = TokenAuthentication(DatabaseJtiCache(workflowEngine.db))
+        val senderName = request.headers[CLIENT_PARAMETER]
+            ?: request.queryParameters.getOrDefault(CLIENT_PARAMETER, "")
+        // todo This code is redundant with validateRequest.  Remove from validateRequest once old endpoint is removed
+        if (senderName.isBlank())
+            return HttpUtilities.bad(request,"Expected a '$CLIENT_PARAMETER' query parameter")
+        val sender = workflowEngine.settings.findSender(senderName)
+            ?: return HttpUtilities.bad(request, "'$CLIENT_PARAMETER:$senderName': unknown sender")
+
+        tokenAuthentication.checkAccessToken(request, "${sender.fullName}.report")
+            ?: return HttpUtilities.unauthorizedResponse(request)
+        return ingestReport(request, context)
+    }
+
+    private fun ingestReport(request: HttpRequestMessage<String?>, context: ExecutionContext): HttpResponseMessage {
```

Comment:
Both the old and new endpoints must be dual active for some time, to give our customers time to transition.  So, both call `ingestReport` where all the real work besides auth is done.

Reply:
Looks good.